File: identhub/src/main/java/de/solarisbank/identhub/di/ActivityModule.java
package de.solarisbank.identhub.di;

import android.content.Context;
import android.content.SharedPreferences;

import de.solarisbank.identhub.base.BaseActivity;
import de.solarisbank.identhub.data.preferences.IdentificationStepPreferences;

import static android.content.Context.MODE_PRIVATE;

public final class ActivityModule {

    private final BaseActivity baseActivity;

    public ActivityModule(BaseActivity baseActivity) {
        this.baseActivity = baseActivity;
    }

    public Context provideContext() {
        return baseActivity;
    }

    public SharedPreferences provideSharedPreferences(Context context) {
        return context.getSharedPreferences("identhub", MODE_PRIVATE);
    }

    public IdentificationStepPreferences provideIdentificationStepPreferences(SharedPreferences sharedPreferences) {
        return new IdentificationStepPreferences(sharedPreferences);
    }
}

File: core/src/main/java/de/solarisbank/sdk/core/di/ActivityModule.kt
package de.solarisbank.sdk.core.di

import android.content.Context
import android.content.SharedPreferences
import de.solarisbank.sdk.core.BaseActivity

class ActivityModule(private val baseActivity: BaseActivity) {
    fun provideContext(): Context {
        return baseActivity
    }

    fun provideSharedPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences("identhub", Context.MODE_PRIVATE)
    }

}